tests/foreman/ui/test_contentview.py

```diff
@@ -644,641 +369,94 @@ def test_positive_update_name(self):
-
-        :id: 715db997-707b-4868-b7cc-b6977fd6ac04
+    @tier1
+    def test_positive_delete(self):
+        """Delete content views
 
-        :setup: content view with custom yum repo
+        :id: bcea6ef0-bc25-4cc7-9c0c-3591bb8810e5
 
-        :expectedresults: Systems can be subscribed to content view(s)
+        :expectedresults: Content view can be deleted and no longer appears in
+            UI
 
-        :CaseLevel: Integration
+        :CaseImportance: Critical
         """
-        cv_name = gen_string('alpha')
-        repo_name = gen_string('alpha')
-        repo_url = FAKE_0_YUM_REPO
-        # create an organization
-        org = entities.Organization().create()
-        # create a lifecycle environment
-        env = entities.LifecycleEnvironment(organization=org).create()
-        # create a yum repository
-        self.setup_to_create_cv(
-            org_id=org.id, repo_name=repo_name, repo_url=repo_url)
         with Session(self) as session:
-            # create content view
-            make_contentview(session, org=org.name, name=cv_name)
-            self.assertIsNotNone(self.content_views.search(cv_name))
-            # add the repository to content view
-            self.content_views.add_remove_repos(cv_name, [repo_name])
-            # publish the content view
-            version = self.content_views.publish(cv_name)
-            self.assertIsNotNone(
-                self.content_views.version_search(cv_name, version))
-            # promote the content view
-            status = self.content_views.promote(cv_name, version, env.name)
-            self.assertIn('Promoted to {}'.format(env.name), status)
-            self.assertIn(env.name, self._get_cv_version_environments(version))
-            # create an activation key
-            content_view = entities.ContentView(
-                organization=org, name=cv_name).search()[0].read()
-            activation_key = entities.ActivationKey(
-                organization=org,
-                environment=env,
-                content_view=content_view
-            ).create()
-            # create a vm host client and ensure it can be subscribed
-            with VirtualMachine(distro=DISTRO_RHEL7) as host_client:
-                host_client.install_katello_ca()
-                host_client.register_contenthost(
-                    org.label, activation_key.name)
-                self.assertTrue(host_client.subscribed)
-                # assert the host_client exists in content hosts page
-                self.assertIsNotNone(
-                    self.contenthost.search(host_client.hostname))
+            for name in valid_data_list():
+                with self.subTest(name):
+                    make_contentview(
+                        session, org=self.organization.name, name=name)
+                    self.assertIsNotNone(
+                        self.content_views.search(name),
+                        'Failed to find content view %s from %s org' % (
```

Comment:
Nitpick, I'd say new style formatting is preferable, several occurrences, up to you